### PR TITLE
GH-564: regression datasets

### PR DESCRIPTION
**Pull request by alanakbik**
This PR fixes a problem with the loss function in the (still beta) `TextRegressor` and adds 4 regression datasets.

You can now train a regression model like this: 

```python
# get the corpus
from flair.datasets import WASSA_SADNESS
corpus = WASSA_SADNESS()
print(corpus)

# choose some word embedding or stack of embeddings
embeddings = OneHotEmbeddings(corpus)

# choose some document embedding
document_embedding = DocumentPoolEmbeddings([embeddings], fine_tune_mode='none')

# init text regressor
classifier = TextRegressor(document_embedding)

# init model trainer
trainer = ModelTrainer(classifier, corpus)

trainer.train(
    f"resources/taggers/test-regression",
    mini_batch_size=4,
    max_epochs=20,
)
```

### CODE REVIEW

**Comment by alanakbik** [comment]
:+1:

**Comment by annguyenthai2784** [comment]
How to predict a sentence after training:
---------------------
from flair.models.text_regression_model import TextRegressor
from flair.data import Sentence
regressor = TextRegressor.load('/flair_text_regression_150/best-model.pt')
sentence = Sentence('@ErinAndrews I ❤️you on DWTS You make my night every show! 😘 #joy ')
scores, _ = regressor.forward_labels_and_loss([sentence])
print(scores.cpu().detach().numpy()[0][0])
------------------------
But I got the following exception: 

0.7307619
/usr/local/lib/python3.6/dist-packages/torch/nn/modules/loss.py:446: UserWarning: Using a target size (torch.Size([0])) that is different to the input size (torch.Size([1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size. return F.mse_loss(input, target, reduction=self.reduction)
